addons/website_event_sale/controllers/main.py

```diff
@@ -70,3 +71,19 @@ def registration_confirm(self, event, **post):
                 request.website.sale_reset()
 
         return res
+
+
+class PaymentPortalOnsite(PaymentPortal):
+
+    def _validate_transaction_for_order(self, transaction, sale_order_id):
+        """
+        Throws a ValidationError if the user tries to pay for a ticket which isn't available
+        """
+        super()._validate_transaction_for_order(transaction, sale_order_id)
+        sale_order_sudo = request.env['sale.order'].browse(sale_order_id).exists().sudo()
+        count_per_ticket = request.env['event.registration']._read_group(
+            [('sale_order_id', 'in', sale_order_sudo.ids), ('state', '!=', 'cancel')],
+            ['event_ticket_id'], ['__count']
+        )
```

Comment:
I'm pretty sure you won't have access to event.registration without `sudo`

I guess maybe the test is succeeding because it's catching an error anyway